ports/stm32/sdcard.c

```diff
@@ -175,7 +175,9 @@ void sdcard_init(void) {
 
     // configure the SD card detect pin
     // we do this here so we can detect if the SD card is inserted before powering it on
-    mp_hal_pin_config(MICROPY_HW_SDCARD_DETECT_PIN, MP_HAL_PIN_MODE_INPUT, MICROPY_HW_SDCARD_DETECT_PULL, 0);
+	#if defined(MICROPY_HW_SDCARD_DETECT_PRESENT)
```

Comment:
I would suggest to use `#if defined(MICROPY_HW_SDCARD_DETECT_PIN)` instead.

Reply:
OK, makes perfect sense. Updated.